backend/controller/dal/fsm.go

```diff
@@ -45,9 +45,16 @@ func (d *DAL) StartFSMTransition(ctx context.Context, fsm schema.RefKey, executi
 		Backoff:           retryParams.MinBackoff,
 		MaxBackoff:        retryParams.MaxBackoff,
 	})
+	observability.AsyncCalls.Created(ctx, destinationState, origin.String(), int64(retryParams.Count), err)
 	if err != nil {
 		return fmt.Errorf("failed to create FSM async call: %w", dalerrs.TranslatePGError(err))
 	}
+	queueDepth, err := d.db.AsyncCallQueueDepth(ctx)
+	if err == nil && queueDepth > 0 {
+		// Don't error out of an FSM transition just over a queue depth retrieval
+		// error because this is only used for an observability gauge.
+		observability.AsyncCalls.RecordQueueDepth(ctx, queueDepth)
```

Comment:
Yeah, it just spams the collector with a lot of 0 logs otherwise. Essentially, this `Record` call only fires when async calls are potentially enqueued. There's a separate `Record` call for when calls are dequeued and executed. So 0s are logged on dequeue but not enqueue because that basically just means this function ran without creating any async calls.

Reply:
Actually thinking about this some more, I might as well not even query for queue depth here unless `successful > 0` to avoid loading the DB unnecessarily. I'll go ahead and change that now.

Reply:
Alright no more need for the `queueDepth > 0` check! And on second thought you were absolutely right that it wasn't necessary for FSM anyway because it would never equal 0 to begin with. So I removed that check from both, and rely on `successful > 0` in pubsub instead.